### DISCUSSION TITLE
Supabase Auth: Changes to default email provider

### DISCUSSION THREAD

**Discussion by kangmingtay** [supabase/supabase#29370] [Changelog]
As our user base has grown, we are taking steps to make sure we are able to continue to provide a safe, secure, robust free plan experience. To ensure that email-based auth continues to work for all users on Supabase, we're making changes if you're using the [default email provider](https://supabase.com/docs/guides/auth/auth-smtp#auth-smtp). This allows us to continue to offer our default provider in a more sustainable and resilient manner.

For maximum flexibility and control over your auth emails, we suggest one of the following:
- Use a [custom SMTP provider](https://supabase.com/docs/guides/auth/auth-smtp#how-to-set-up-smtp) instead
- Send emails through your own email provider using the [email send hook](https://supabase.com/docs/guides/auth/auth-hooks/send-email-hook)

If you still want to use the default email provider, these are the changes being planned:

- **Email template customization will be allowed and customized email templates will not be reverted to default.**

- **26th September**: If you do not have a custom SMTP server set up, emails can only be sent to email addresses in your project's organization. So for example, if your organization has the following members: person-a@example.com, person-b@example.com and person-c@example.com , this means that email messages from Auth will only be sent to these addresses. 

These measures are taken to prevent abuse to our shared SMTP service. In the future, we may consider increasing the email rate limits once we see a drop in abuse.

### Why such a short notice?

Supabase uses a third-party email sending provider that has mandated we reduce email abuse significantly or they will be forced to block all email sending. A [tragedy of the commons](https://en.wikipedia.org/wiki/Tragedy_of_the_commons).

### Can't Supabase switch to a different email sending partner?

Yes, but we would run into the same issues. All email sending services are required to monitor abuse and force their customers to follow the same rules.

### Can't Supabase send emails on their own, without a third party?

Not really. You can't just send email on the web today without investing a lot of money and time (unblocking port 25, keeping IP addresses out of spam lists, etc.). This is not our core competency and do not have plans to start doing this today.

### How long does it take to set up a custom SMTP provider?

Fortunately this is very easy. You can use any email sending service for this, really popular ones include:

- [Resend](https://resend.com/)
- [AWS SES](https://aws.amazon.com/ses/)
- [Postmark](https://postmarkapp.com/)
- [SendGrid](https://sendgrid.com/)
- [Zepto Mail](https://www.zoho.com/zeptomail/)
- [Brevo](https://www.brevo.com)

All you need to do is create an account, verify your sending domain and finally input the SMTP username and password in the [Auth settings page](https://supabase.com/dashboard/project/_/settings/auth).

### What if I turn off email confirmations, can I use it then?

_Currently this behavior is not supported and we'll be rolling out a fix for it during the first week of October._

Confirming email addresses is where most of the email message activity for a project originates. Turning it off can be a viable option for some projects that are still in the early testing, development or experimental phase. 

Be aware that even if you turn off email confirmations the forgot password or reset password flows in your app continue to function. They also send messages, and starting 26th September those messages will be delivered only to the members of the Supabase organization that owns the project. All other end-users will get a message similar to "Email address not authorized." Effectively, the forgot password / reset password flow will be broken for your project.

### What if I want just username + password authentication and using `<username>@<fakedomain>` instead?

Please don't do this. Part of the reason why we were forced to lock down these changes is bounced emails, probably from use cases like this.

Official username + password support is going to be made available in the coming year, and until then:

- Use a real domain, that you control
- Send emails to that domain, so set up a receiving server

But the best thing to do is:

- Set up a [Send Email Auth Hook](https://supabase.com/docs/guides/auth/auth-hooks/send-email-hook) that does nothing. You don't even need to use a server or an Edge Function. Just define a Postgres function that just does nothing.

### I'm using the admin API to generate links, and not really sending using Supabase's default provider. Do I need to do anything?

_All projects using generate link via the admin API without custom SMTP have been patched to allow the behavior. We still strongly urge those customers to set up custom SMTP regardless._

Just because you're mostly using the admin API to generate links to send in custom email messages, doesn't mean that the Auth server is not configured to use Supabase's shared SMTP service. Your Auth API can be called from your frontend at any time, especially in edge cases such as to handle forgot password or other similar flows, which you may not be handling via the admin API.

Therefore we urge all customers that do use the admin API to set up a custom SMTP sending service regardless.

If you are not interested in setting this up, you can instruct the Auth server to ignore all emails (pretend it's sending them) by configuring a [Send Email Auth Hook](https://supabase.com/docs/guides/auth/auth-hooks/send-email-hook) as a Postgres function that does nothing.

### How can I disable the warning banner?

You can disable the warning banner by setting up a [custom SMTP provider](https://supabase.com/docs/guides/auth/auth-smtp#how-to-set-up-smtp) , or, if your project doesn't use email at all, by disabling the [email provider](https://supabase.com/dashboard/project/xguihxuzqibwxjnimxev/auth/providers).

### [20th September 2024](https://github.com/orgs/supabase/discussions/29370#discussioncomment-10702421)

**Email template customization will be allowed and customized email templates will not be reverted to default.**

Team has decided that restricting email template customization is not viable and a big breaking change. We may need to do go back to this in the future if abuse continues and our other measures like allowing projects to only send messages to authorized email addresses do not improve the situation. We continue to urge all customers regardless of plan that use the default SMTP service for live applications to move to a custom SMTP provider as soon as able.

- ~**20th September**: [Email template customization](https://supabase.com/dashboard/project/_/auth/templates) will no longer be possible without setting up a custom SMTP provider. Email templates already customized can still be customized until 24th September.~

- ~**24th September**: Projects without a custom SMTP provider will have their custom email templates returned back to the default ones from Supabase. This means that any auth emails sent out from your project will use the default email template.~

**Comment by echarrod** [comment]
Appreciate the reasoning behind the changes 👍 

It does feel like a short timeline though, in order for us to implement a custom SMTP server, so we can send emails outside of our org 😒

**Reply by kangmingtay** [reply]
yeah it was a tough decision to make since the risk is our email sending partner shutting down our service which will prevent everyone from sending any emails

fortunately, setting up a custom SMTP service is quite easy and straightforward - my preference is to use [Resend](https://supabase.com/partners/integrations/resend) which we have an integration with

**Comment by tanoabeleyra** [comment]


**Reply by hf** [reply]
> Why such a short notice?

We do not send emails ourselves, so our email sending partner is forcing our hand to significantly restrict abuse or they are going to block us from sending any emails. 

Had we had a choice, we wouldn't have done it in such a short notice.

**Reply by khera** [reply]
While you can differentiate this kind of usage pattern, it doesn't protect a service provider in reality. People who abuse systems will find any way they can to automate the "let me convince them I'm a legitimate user" then flip over to abusing it once their limits are lifted. It is especially bad when the service is free of cost.

The only thing stopping you personally from suddenly sending hundreds of messages using a customized confirmation message is that you are acting in good faith.

**Comment by GaryAustin1** [comment]
Also I and other users on Discord use [Brevo](https://www.brevo.com/landing/smtp/) .
It has a decent free plan.

**Reply by khera** [reply]
For those who are not inclined to use a free service, Zoho has a service called zeptomail that is dirt cheap and pretty good at delivery.

**Reply by MarkOSullivan94** [reply]
Are there any cons to be aware of before using them?

**Comment by marcsasson** [comment]
Does this only affect free plans? If I upgrade will I not be affected?

**Reply by GaryAustin1** [reply]
Pro plan users are getting emails (warnings about this) and the wording/description here sounds like it applies to all plans.

**Reply by marcsasson** [reply]
> robust free plan experience.

That's why I though it was only affecting free plans.

Also, is the abuse mostly coming from free plans? If yes, maybe give pro plans more time to migrate?

**Reply by GaryAustin1** [reply]
Have to wait for clarification from SB then.  But they have always recommended that in production you do not use their email service.  Of course that was mainly focused on the 3 to 4 email hour limit.

**Reply by J0** [reply]
Hey,

Thanks for the query. Unfortunately, this affects all plans. Free plan users have a great percentage of projects without custom SMTP and will probably be most significantly affected by this change. In general, most folks on paid plans tend to have set up Custom SMTP as part of the [going into production](https://supabase.com/docs/guides/platform/going-into-prod) guide.

**Reply by hf** [reply]
> Also, is the abuse mostly coming from free plans? If yes, maybe give pro plans more time to migrate?

@marcsasson Turns out most of the abuse actually comes from paid plans that have not moved to custom SMTP, as those projects tend to be live and have a user base or are prone to abuse themselves.

**Comment by sheprograms** [comment]
Since I'm being forced to use SendGrid, I'd strongly prefer to use their dynamic email templates instead of Supabase's. Can you recommend any step-by-step documentation for setting up an Email Hook with the SendGrid API? I'm mostly struggling with figuring out how to pass custom parameters. My use case is to send invites to join a team. I'd like to include the logo of the team in the email. My dynamic template is set up to accept this parameter, but I need to pull it from the teams table in supabase before hitting the sendgrid API with all of the relevant confirmation links and tokens.

**Reply by kangmingtay** [reply]
@sheprograms have you seen [this guide](https://supabase.com/docs/guides/auth/auth-hooks/send-email-hook)? you can swap out the places where resend's api is called with sendgrid's

**Comment by hf** [comment]
Hey everyone, update from today.

**Email template customization will be allowed and customized email templates will not be reverted to default.**

Team has decided that restricting email template customization is not viable and a big breaking change. We may need to do go back to this in the future if abuse continues and our other measures like allowing projects to only send messages to authorized email addresses do not improve the situation. We continue to urge all customers regardless of plan that use the default SMTP service for live applications to move to a custom SMTP provider as soon as able.

During the day today the platform will allow you to customize email templates.

**Comment by Sergio-P** [comment]
I used custom templates to translate the text of the official ones to Spanish. Have you considered adding support for other languages in the templates? It’s difficult to develop for non-English speakers without this feature.

**Reply by GaryAustin1** [reply]
@Sergio-P there is now an auth hook to do things like that. https://supabase.com/docs/guides/auth/auth-hooks/send-email-hook

**Comment by adeniranbukunmi** [comment]
please could this be the reason why my Django application keep saying 500: INTERNAL_SERVER_ERROR on vercel ? after supplying my environment variables

**Reply by kangmingtay** [reply]
nope, it should still work - check your project's auth logs [here](https://supabase.com/dashboard/project/_/logs/auth-logs) and if it's still an issue, please open a ticket at https://supabase.help so we can investigate

**Comment by nukeop** [comment]
I don't mind paying for an email service but I do mind being de-anonymized and doxed, and so many of those providers required actual, real documents which is a huge invasion of privacy. Can I just pay you more to have this go away? You must have info on who's sending spam (because what other email abuse can there be). Very unsportsmanlike for them to give you only such a short notice which you then have to pass on to your clients. In general I'd rather pay Supabase a premium for handling emails so that I don't have to worry about one more third party provider.

**Reply by hf** [reply]
> You must have info on who's sending spam (because what other email abuse can there be).

This is actually quite hard to figure out. From our analysis essentially all customers use Auth emails for their right purpose. However, the "spam" actually comes from unwanted email or in relation to projects that are operating in a "spammy" industry. 

We don't want to police customers and set up complex review rules. Anyone can build apps on Supabase, but to send email unfortunately you will need to do some extra work.

> Can I just pay you more to have this go away?

At this time we don't plan to offer this as a service, but given how important this is we may be looking at other alternatives such as partnering with some companies to ease the burden. No concrete plans yet.

**Reply by nukeop** [reply]
Thanks! Supabase is a fantastic service, with outstanding support.

**Comment by alphanumericaracters** [comment]
Today I open my dashboard and I found the message, I check the list of emails sent and there were only 5: (mom, dad, my wife, my daughter and my dog ​​[who has a service role]). 

Aand I said... 'WTF Supabase!' This is not abuse at all! 🤷‍♂️

Then I read this.  😁

There will be no choice but to make a script in a Google account, which acts as a webhook, some triggers to communicate the creation of a new user and token, the email is sent with a magicklink that redirects to the App and blah blah blah...

(☞ﾟヮﾟ)☞    Argentinian way 😎 The **Favaloro** Script [bypass master]  ☜(ﾟヮﾟ☜)

**Comment by enriquesalomon** [comment]
I am not using the auth email sending confirmation for the created user account in my app. I just turn off the confirmation of the email. Does it resolve the issue? since my app , was the only admin account can create or signup a user account.  My question is, it will still affect to my app? or it will not.

**Reply by hf** [reply]
Turning off email confirmation gets you there 80% of the way. However be mindful that when you use email + password in your app, the forgot password / reset password flow still requires for emails to be sent.

On 26th of September only the people part of your Supabase organization will be able to effectively reset their passwords, all others will get an error message "Email not authorized."

**Reply by EmePin** [reply]
> Turning off email confirmation gets you there 80% of the way. However be mindful that when you use email + password in your app, the forgot password / reset password flow still requires for emails to be sent.
> 
> On 26th of September only the people part of your Supabase organization will be able to effectively reset their passwords, all others will get an error message "Email not authorized."

thanks

**Comment by muchirajunior** [comment]
First how can i get rid of this warning off my view, i don't even send emails on supabase

**Reply by hf** [reply]
Disabling the email provider will get rid of the warning.

**Comment by damiensedgwick** [comment]
Does this include Magic Link sign ins? So going forward I would need to setup a provider for logging in?

**Reply by j4w8n** [reply]
> Does this include Magic Link sign ins? So going forward I would need to setup a provider for logging in?
> 

Yep

**Reply by damiensedgwick** [reply]
No dramas, just wanted to double check.

**Comment by alok-mishra143** [comment]
i am a free plan user and i don`t have custom domain also how can i use custom smtp

**Reply by J0** [reply]
Hey @alok-mishra143,

Rest assured that you don't need the custom domain add-on to set up custom smtp. You can find instructions for [configuring a custom SMTP here](https://supabase.com/docs/guides/auth/auth-smtp)

**Reply by J0** [reply]
I realized that you might be referring to the sender email. There are providers which don't require purchase of a domain iirc (e.g outlook,gmail). It's also possible to find free domains though I would recommend buying one if within budget.

**Comment by samifouad** [comment]
Was planning on integrating with [Postmark](https://postmarkapp.com), as I only plan to use e-mail auth for my project. I guess my timeline to make that happen is now bumped up. Bad situation for everyone not prepared for this, but it is what it is. Thanks for the communication on this, that's all we can ask for when situations like this arise.

**Reply by hf** [reply]
We expect that most projects that were serious about email sending already have set up custom SMTP, as the existing rate limit was around 2 messages per hour.

**Comment by d0nda** [comment]
the integration with Resend was easy and straight forward it didnt take me more than 5 minutes to setup. ⚡

**Comment by Yaodeer** [comment]
I hastily saved the default password when setting up the custom SMTP, but later, after generating the correct password in Gmail, I found that I could no longer change the default password because it indicates that it cannot be changed after saving. What should I do?💔

**Reply by hf** [reply]
Oh I see what you mean. You can change it, the UI doesn't show it again because it's stored encrypted in our database. I can try and get this fixed in the next few days if it's too awkward.

**Reply by Yaodeer** [reply]
Thank you very much for your reply! It seems that my problem was entering the wrong username. I thought I didn’t need to enter the Gmail address again in that field, so I just entered my Gmail username instead😢

**Comment by hf** [comment]
Restrictions are rolling out in the next few minutes. It should be applied to all projects by end of day.

Thank you for cooperating with us on this urgent change, we wish we didn't have to do this!

**Comment by victorfriedrich** [comment]
It seems Resend requires a domain for the integration, is there an option to do this with the auto-generated supabase / vercel ones for early prototypes?

**Reply by hf** [reply]
Don't think so, but check the Resend docs.

**Comment by richleach** [comment]
I'm on the free/hobby account, is there anyway I can see my log files? I keep getting "Error sending confirmation email" but can't see any more details than that. BTW I set up with Brevo, pretty easy I guess. I wish I'd seen your note about Resend though

**Reply by GaryAustin1** [reply]
Goto the logs section of the dashboard and then Auth log.  There might be a bit more info on the error back from the provider if that is the issue.  Also make sure your rate limit is set higher than 3.  I use Brevo and have had no issues.

**Reply by richleach** [reply]
Rate limit?

**Reply by GaryAustin1** [reply]
![image](https://github.com/user-attachments/assets/9089097d-8982-4898-b971-7b4108506eba)

**Reply by richleach** [reply]
Thank you Gary, still getting used to the UI

**Comment by evguu** [comment]
I see no problem with this as I don't send the emails, however this also completely prevents me from creating fake users for testing purposes from the UI.
![User creation menu](https://github.com/user-attachments/assets/30b82d11-b72f-4fca-8321-7d53ea80ee0b)

I don't see why this would be blocked since 'Creating users via this form does not send email'.

Attempts to create user fail with the message 'Failed to create user: Email address "a@a.com" cannot be used as it is not authorized'.

Could you look into this or give advice on a different way to create test users? And no, creating custom SMTP for such simple testing that **does not require email sending** is not viable.

**Reply by Kacper1098** [reply]
I am facing the same problem,  my e2e tests create user accounts which are fake emails and I keep getting `email_address_not_authorized` error.

**Reply by hf** [reply]
Thanks for letting us know about this, yes this is a slight oversight and we'll fix it next week (first week of October).

**Reply by byrnehollander** [reply]
@kangmingtay @hf This change appears to have impacted the [`generateLink` function](https://supabase.com/docs/reference/javascript/auth-admin-generatelink); the docs say that this function was designed for use with custom email providers.

I am generating magic links on my server and sending them through Mailgun, but I was getting the following error that was preventing users from logging in:

```
Email address "someUser@gmail.com" cannot be used as it is not authorized
```

I was able to toggle on `Enable Custom SMTP` and enter fake credentials to get around this.

But I imagine that this is impacting actual customers in production (like me), and not just breaking people's tests.

---

_Edit_: 

I believe this was introduced in https://github.com/supabase/auth/pull/1778/files#diff-97dfc3310e5139b4d9c7a160bd077aeb81ca1e7b8d7cfa837a402f54b4c03a3d

Some evidence: the above error message is not in the codebase until it is added in this PR in [`internal/api/mail.go, line 578`](https://github.com/supabase/auth/pull/1778/files#diff-1bee429deb4821de838762a5cce72a874991d044408007a85e1f988e59e0c0c4L578)


I'm not certain this will fix the issue, but I put up a PR here: https://github.com/supabase/auth/pull/1779

**Reply by hf** [reply]
There is a slight misconception here. Yes there's a bug where the admin API shouldn't have been affected by the change, that's clear and getting fixed.

However, it's not true that if you use the admin API to generate links for email messages that you are not using the built-in SMTP provider. Your project's APIs are live and open to the world to call, so your application could be calling the forgot password flow without using the admin link.

This is not entirely obvious, so I urge all customers facing the same issue to set up custom SMTP _regardless of using the admin APIs._ If you really don't want any emails to be sent, it's best to setup a Send Email Auth Hook that is a no-op Postgres function instead.

**Reply by hf** [reply]
All projects that have used the generate link admin API recently have been patched. Fix is rolling out to all projects in the next week.

**Reply by byrnehollander** [reply]
Thank you @hf. I understand that there are a lot of moving parts here and you had to work under a short deadline.

---

> I urge all customers facing the same issue to set up custom SMTP _regardless of using the admin APIs_. If you really don't want any emails to be sent, it's best to setup a Send Email Auth Hook that is a no-op Postgres function instead.

Is there another way to disable emails from being sent? I feel like there could just be a toggle built into the product.

> All projects that have used the generate link admin API recently have been patched. Fix is rolling out to all projects in the next week.

Can you share more about how changes get rolled out to prod? This bug impacted me at Friday night, which was pretty bad timing. 

Do bug fixes roll out faster than feature releases? "in the next week" seems like a very long time horizon for a bug fix that can break login

**Reply by Frikster** [reply]
Am brand new to Supabase and I just bumped into this. If you run a services that does email as a service in any way test accounts are absolutely essential so this is pretty annoying. Am very glad to hear it will be fixed in the next week.

**Comment by davidd8** [comment]
Based on the timing of this change, it may have affected users with already configured custom SMTP servers. My service is seeing 500s and 504s for sending magic links, and users are unable to log in.

* For verified users: `500: Error sending magic link email","path":"/otp" - unexpected_failure, AuthApiError`
* For new, unverified users: `504: gateway timeout  AuthRetryableFetchError <> undefined`

In my case, I have Amazon SES, and no outgoing emails have been sent since this afternoon. My hunch is these errors are preventing the email from being generated in the first place, as noted by @byrnehollander.

**Reply by hf** [reply]
This is unlikely to be related to the changes. Reach out to https://supabase.help, check your logs and SES integration.

**Reply by davidd8** [reply]
I did file a ticket, but I think it is related, as I changed nothing in my setup since my last post and retried today and it all works again. There seems to have been a regression in the rollout, which was reverted?

**Reply by davidd8** [reply]
I spoke too soon, the issue has been resolved for NEW users, but EXISTING users are still seeing 500s like this: 

`"auth_event":{"action":"user_recovery_requested"... actor_via_sso":false,"log_type":"user"},"component":"api","error":"html/template:https://api.supabase.com/platform/auth/<app-id>/templates/magic-link: "\"" in attribute name: "\"\u003eMacOS\u003c/a\u003e) \nand Mobile (\u003ca hre"","level":"error","method":"POST","msg":"500: Error sending magic link "`

**Reply by kangmingtay** [reply]
@davidd8 that's seems like an issue with your email template

**Reply by davidd8** [reply]
Oh wow, great catch @kangmingtay you are correct, I had a typo in one of the email templates - user error! Problem solved :)

**Comment by sean-dev-only** [comment]
I am not using the email approach. I am using notion provider's 

```
const {data, error} = await supabase.auth.signInWithOAuth({
    provider: 'notion',
    options: {
      redirectTo: someUrlWithSearchParams?a=a&b=b`,
    },
  })
```
I checked the log, it says 
```
"error":"oauth2: \"invalid_client\"","level":"error","method":"GET","msg":"500: Unable to exchange external code: xxxx-3c18-xxxx-xx-xx"
```  
The error starts to show this afternoon too

**Reply by J0** [reply]
Hey @sean-dev-only,

Thanks for the query -  the change shouldn't affect OAuth sign ins as they don't involve emails. This looks like a separate error, related to the exchange of the  Auth code between Auth and notion.

Could we trouble you to double check the Client ID and Secret in use?

Thanks

**Reply by sean-dev-only** [reply]
@J0 
Thanks for the reply. 
I want to close my thread by this https://status.notion.so/incidents/ch2cmpln7jjg
it's on Notion. Sry for the distraction.

**Comment by Juiciator** [comment]
Hi, I'm not a tech guy, we use supabase for a flutterflow app. I get the message that I need to configure smtp every time I log in in supabase, but we don't actually send emails through supabase, and our accounts are google workspace. Could you please disable that message for my account?
thanks

**Reply by Juiciator** [reply]
![another button](https://github.com/user-attachments/assets/6c0e47d3-dea0-4291-9469-2fe857561d34)
May be you can add this button to that option

**Reply by hf** [reply]
You can disable the email provider on this page which will remove the message: https://supabase.com/dashboard/project/_/auth/providers

**Reply by hf** [reply]
We will add a button to help you turn it off soon, but at this time we want to increase awareness of this change for all customers that are affected.

**Comment by Juiciator** [comment]
All the users of my app login using google workspace, the question is 
¿To disable the Email provider in supabase will allow me to login in my app using google auth login?
thanks in advance for the answer

**Reply by hf** [reply]
Yes, using Social Login / OAuth with Google, Apple, or any other such provider remains unaffected.

**Comment by sidharthhhh** [comment]
![Screenshot 2024-09-28 172326](https://github.com/user-attachments/assets/2040aea9-e369-4fc9-b3d6-f0fbd9ca28e2)

while created user mannually, i am only able to created one user while my own gmail, with other gmail id , i am not able to create any user. How can i allow that that with all email i can create user?

**Reply by hf** [reply]
Please open a support ticket at https://supabase.help for us to help you most effectively.

**Reply by sidharthhhh** [reply]
thanks sir

**Comment by alignbenpersitz** [comment]
Is this impacting existing users? I'm getting an error when trying to login now that my email address is not authorized. We aren't sending any emails, and don't require confirmation of accounts. These emails have worked previously, but now I can't login with them. 

Emails that are members of the project CAN login even though they're listed as the same domain of the users who cannot login.

**Reply by Jacksonant** [reply]
+1

**Reply by Cerebrolabs** [reply]
Same. Wtf is happening?

**Reply by ScriabinOp8No12** [reply]
+1 Same problem

**Reply by ryanhayame** [reply]
I was also unexpectedly affected today on a website that does not send emails or use email confirmation. Users could not signup or signin. I ended up just following this https://resend.com/docs/send-with-supabase-smtp to solve the issue quickly. I'm confused because I think effectively nothing changed since I'm not sending any emails anyways. Even more confusing: when following the steps I verified a completely separate domain (did not have quick access to DNS records of the real one at the time) and it still worked. Not sure why we were affected.

**Reply by matthewharwit** [reply]
+1

**Reply by eckhartt** [reply]
+1 same issue

**Reply by 100kristine** [reply]
+1

> What if I turn off email confirmations, can I use it then?
> Currently this behavior is not supported and we'll be rolling out a fix for it during the first week of October.

Wondering what the rough eta is looking like now? Since it's already Oct 3?

**Reply by 100kristine** [reply]
Just saw @ryanhayame's solution (thank you!!) - adding resend worked for me, even with emails turned completely off

@kangmingtay it might be helpful to have a quick note about the workaround in that part of the post? Since it is only mentioned right now for projects with emails on

**Reply by deepanshubajaj111** [reply]
Faced same issue this morning, and saw a banner on my Supabase dashboard saying "To fight abuse, emails sent by Auth are restricted to the members of this project's organization. Set up a custom SMTP provider to send messages to any user."

taking this in charge, I added a new team member in my org and WLA! I was able to login with that email, I guess this is due to new SMTP policies, and the best solution to this is to setup custom SMTP.

**Reply by MonoliYoda** [reply]
The problem is that both the announcement and the banner only mentioned sending emails, nothing about blocking users from logging in.

**Reply by MitchelliJ** [reply]
Yep, same here. Woke up getting calls from customers not being able to log in. I am not using any mails but apparently all users from outside the associated 'organisation' are unauthorised to log in all of a sudden. Not cool.
I tried adding Resend to SMTP but some users are still not able to log in. Others are. No idea whats happening.

**Reply by j4w8n** [reply]
> +1
> 
> > What if I turn off email confirmations, can I use it then?
> > Currently this behavior is not supported and we'll be rolling out a fix for it during the first week of October.
> 
> Wondering what the rough eta is looking like now? Since it's already Oct 3?

Looks like they just merged a fix. However, I don't know how quickly that rolls out to cloud instances.

https://github.com/supabase/auth/releases/tag/v2.162.1

**Reply by deepanshubajaj111** [reply]
I guess the issue is fixed, I'm able to login without even setting up SMTP

**Reply by ScriabinOp8No12** [reply]
> I guess the issue is fixed, I'm able to login without even setting up SMTP

Edit: Partially working for me now, existing signed up users can sign back in, but a new user can't signup like before.

**Reply by EvdB1975** [reply]
Same here. Existing users can log in, but new users can't register. I'm not using email confirmation. I'm using the database as a backend for a school project/research. When trying to register a new user, I get the 'email_address_not_authorized' error.

**Reply by ScriabinOp8No12** [reply]
> Same here. Existing users can log in, but new users can't register. I'm not using email confirmation. I'm using the database as a backend for a school project/research. When trying to register a new user, I get the 'email_address_not_authorized' error.

I see, good to know, thanks.  We are using Supabase in production, so it's not very optimal for this to happen lol

**Reply by enriquesalomon** [reply]
> Same here. Existing users can log in, but new users can't register. I'm not using email confirmation. I'm using the database as a backend for a school project/research. When trying to register a new user, I get the 'email_address_not_authorized' error.

have you tried setting an SMTP ? try resend, it will helps solving that problem, just try to add that and then turning off the auth email. It works at my side, it was the same scenarios as yours.

**Reply by hf** [reply]
> I guess the issue is fixed, I'm able to login without even setting up SMTP

Yes unfortunately we were too aggressive with it but realized quickly and fixed as soon as we could. Sorry about that!

**Comment by AdditionAddict** [comment]
Getting hit hard by this this morning, please add instructions for Mailgun 😭

**Reply by hf** [reply]
Please visit this page: https://documentation.mailgun.com/docs/mailgun/user-manual/sending-messages/#send-via-smtp

**Comment by nukeop** [comment]
Everything works on my end, before and after using Resend for SMTP, even though I did it after the deadline. Supabase is the superior choice as always.

**Reply by maagyei04** [reply]
please help with it, don't get it

**Comment by AndersonMamede** [comment]
My project was also affected and was only fixed 20 minutes ago after I found the workaround mentioned above, set up Resend as a custom SMTP provider, and finally made new user registrations possible again. However, wasn't this supposed to be only about sending emails, or did I miss the information indicating that it would also prevent new user registrations?

**Reply by hf** [reply]
If you call `supabase.auth.signUp()` unfortunately that counts as sending an email, as we are not able to distinguish between customers that have already sent a link and those that just call this to send a link.

**Comment by ErwinAI** [comment]
I've just added a fake SMTP to allow account creation again. Same situation as others: I don't use Supabase to send emails.

If you are **sure** that you don't rely on any emails being sent from Supabase, just fill in: smtp.gmail.com with random username/password.

**Reply by hf** [reply]
You can instead add a Send Email Auth Hook with a Postgres function that does nothing.

**Comment by Amirshp75** [comment]
I use gmail smtp with random username and password.
But when creating a new user, I receive the following message:
`Exception: Failed to sign up: AuthException(message: {"code":"unexpected_failure","message":"Database error saving new user"}, statusCode: 500, errorCode: null`
I am not technical, please explain the suggested method of Send Email Auth Hook with a Postgres function by mentioning the steps.
thanks.

**Comment by felipesnts** [comment]
To use our own SMTP provider we need to be able to whitelist where the connection is coming from. Are the IP addresses used by Supabase for this documented anywhere? I was not able to find anything related to that. Thanks

**Reply by w3b6x9** [reply]
unfortunately, we do not expose IP addresses that you can add to your smtp provider's allowlist.

**Comment by kxvich** [comment]
we should be able to sign up new users after turning off email confirmation for testing purposes
